src/main/kotlin/no/nav/familie/ef/sak/service/steg/StatusPåOppdragSteg.kt

```diff
@@ -0,0 +1,32 @@
+package no.nav.familie.ef.sak.service.steg
+
+import no.nav.familie.ef.sak.repository.domain.Behandling
+import no.nav.familie.ef.sak.service.TilkjentYtelseService
+import no.nav.familie.kontrakter.felles.oppdrag.OppdragStatus
+import org.springframework.stereotype.Service
+import java.util.*
+
+
+@Service
+class StatusPåOppdragSteg(private val tilkjentYtelseService: TilkjentYtelseService) : BehandlingSteg<Void?> {
+
+    override fun utførSteg(behandling: Behandling, data: Void?) {
+        tilkjentYtelseService.hentStatus(behandling).let {
+            when (it) {
+                OppdragStatus.KVITTERT_OK -> TODO()
+                else -> {
+                    prøvHentStatusPåNytt(status = it, behandingId = behandling.id)
```

Comment:
Hadde man kunnat haft en "integrasjonstest" for tasken som "kaller denne" hvor man sjekker att det blir kastet en feil?

Eks

```kotlin
package no.nav.familie.ef.sak.task

import io.mockk.every
import io.mockk.mockk
import no.nav.familie.ef.sak.no.nav.familie.ef.sak.repository.behandling
import no.nav.familie.ef.sak.repository.domain.Behandling
import no.nav.familie.ef.sak.repository.domain.Fagsak
import no.nav.familie.ef.sak.repository.domain.Stønadstype
import no.nav.familie.ef.sak.service.BehandlingService
import no.nav.familie.ef.sak.service.TilkjentYtelseService
import no.nav.familie.ef.sak.service.steg.StatusPåOppdragSteg
import no.nav.familie.ef.sak.service.steg.StegService
import no.nav.familie.kontrakter.felles.oppdrag.OppdragStatus
import no.nav.familie.prosessering.domene.Task
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class StatusPåOppdragTaskTest {

    val tilkjentYtelseService = mockk<TilkjentYtelseService>()
    val behandlingService = mockk<BehandlingService>()
    val stegService = StegService(listOf(StatusPåOppdragSteg(tilkjentYtelseService)), behandlingService, mockk())
    val task = StatusPåOppdragTask(stegService, behandlingService)

    lateinit var behandling: Behandling

    @BeforeEach
    internal fun setUp() {
        behandling = behandling(Fagsak(stønadstype = Stønadstype.OVERGANGSSTØNAD))
        every { behandlingService.hentBehandling(any()) } returns behandling
    }

    @Test
    internal fun `KVITTERT_OK er ikke implementert ennå`() {
        every { tilkjentYtelseService.hentStatus(any()) } returns OppdragStatus.KVITTERT_OK
        assertThat(Assertions.catchThrowable { task.doTask(Task(StatusPåOppdragTask.TYPE, behandling.id.toString())) })
                .isInstanceOf(NotImplementedError::class.java)
//TODO sjekk att den oppretter ny task for neste steg? Eller att en unittest for steget gjør det? 
    }

    @Test
    internal fun `KVITTERT_FUNKSJONELL_FEIL feiler tasken`() {
        every { tilkjentYtelseService.hentStatus(any()) } returns OppdragStatus.KVITTERT_FUNKSJONELL_FEIL
        assertThat(Assertions.catchThrowable { task.doTask(Task(StatusPåOppdragTask.TYPE, behandling.id.toString())) })
                .isInstanceOf(IllegalStateException::class.java)
    }
}
```

Reply:
Nå når jag ser på dette tenker jag att det hadde vært fint hvis selve tasken hentet status, og satte ny status på tasken beroende på hvilket resultat den fikk. Eks så er det kun LAGT_PÅ_KØ som vi trenger å rekjøre, de andre burde settes til feilet/manuell behandling direkt, i stedet for å feile flere ganger.

Men dette(å sette ny status) er ikke mulig i prosessering slik det er nå

Reply:
Ja det hade vart ønskligt. Men som testen ær satt upp nu ☝️  så testar man enbart att error funktionen kastar IllegalStateException 🤔